Library/Homebrew/requirements/gpg_requirement.rb

```diff
@@ -0,0 +1,27 @@
+require "requirement"
+
+class GPGRequirement < Requirement
+  fatal true
+  default_formula "gnupg2"
+
+  satisfy(:build_env => false) { gpg2 || gpg }
+
+  # MacGPG2/GPGTools installs GnuPG 2.0.x as a vanilla `gpg` symlink
+  # pointing to `gpg2`, as do we. Ensure we're actually using a 2.0 `gpg`.
+  # Temporarily, only support 2.0.x rather than the 2.1.x "modern" series.
+  def gpg
+    which_all("gpg").detect do |gpg|
+      gpg_short_version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
+      next unless gpg_short_version
+      Version.create(gpg_short_version.to_s) == Version.create("2.0")
+    end
+  end
+
+  def gpg2
```

Comment:
Yeah, I ended up dropping 1.x support because 1) Upstream is trying to discourage people from using it unless they actively need something extremely portable, 2) Very few things should ever specifically need 1.x, 3) We're renaming the executables in the top-level prefix for `gpg` & I didn't want to start plonking `PATH` env scripting into every formula.

No issue with calling the requirement `GPG2Requirement`. Will do that. IRT the sharing code, 👍 depending on what we do with the discussion in the other PR around this logic needing replicating in two different places currently.